src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -68,5 +69,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
```

Comment:
Not sure if I understand that. We will have to override that one in the QueryBuilders that we refactor, or am I wrong?